lib/src/query/property.dart

```diff
@@ -0,0 +1,227 @@
+part of query;
+
+class PropertyQuery {
+  Pointer<Void> _cProp;
+  int _type;
+  bool _distinct;
+
+  PropertyQuery(Pointer<Void> cQuery, int propertyId, int obxType) {
+    this._type = obxType;
+    _cProp = checkObxPtr(bindings.obx_query_prop(cQuery, propertyId), "property query");
+  }
+
+  void close() {
+    checkObx(bindings.obx_query_prop_close(_cProp));
+  }
+
+  get distinct => _distinct;
+
+  set distinct (bool d) {
+    _distinct = d;
+    checkObx(bindings.obx_query_prop_distinct(_cProp, d ? 1 : 0));
+  }
+
+  int count() {
+    final ptr = allocate<Uint64>(count: 1);
+    try {
+      checkObx(bindings.obx_query_prop_count(_cProp, ptr));
+      return ptr.value;
+    }finally {
+      free(ptr);
+    }
+  }
+
+  Pointer<TStruct> _curryWithDefault<TStruct extends Struct, N extends NativeType>
+      (obx_query_prop_find_t<TStruct, N> findFn, Pointer<N> cDefault, String errorMessage) {
+    try {
+      return checkObxPtr(findFn(_cProp, cDefault), errorMessage);
+    }finally {
+      if (cDefault.address != 0) {
+        free(cDefault);
+      }
+    }
+  }
+}
+
+/// shared implementation, hence mixin
+mixin _CommonNumeric on PropertyQuery {
+
+  double average() {
+    final ptr = allocate<Double>();
+    checkObx(bindings.obx_query_prop_avg(_cProp, ptr));
```

Comment:
AFAICT this should be inside try-catch because `ptr` is already allocated and must be freed even if checkObx throws

Reply:
also check other instances of this pattern

Reply:
Done. Though still `Attempt to free invalid pointer 0x8` presumably for `test/query_property_test.dart: .sum integers`. Knowing the exact line would be nice...

Reply:
OK, not closing the property query pointer helps. Investigating.

Reply:
Updating here, too: closing seems to work now, for whatever reason.